datanode/sqlstore/sanitize.go

```diff
@@ -0,0 +1,102 @@
+package sqlstore
+
+import (
+	"encoding/hex"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// nolint:nakedret
+func SanitizeSql(sql string, args ...any) (output string, err error) {
```

Comment:
```suggestion
func SanitizeSQL(sql string, args ...any) (output string, err error) {
```

Reply:
What's  the reason to add this and for not needed it before in general btw? I may not see the whole picture, but the input coming always from the core I'd expect them to be pretty much always well formatted and it to be a bug if not?

Reply:
The newly added `CopyTo` method doesn't take args.

Passing the parameters to the SQL query such as `partnerID`, `assetID` and so on, comes from the API request and ends up as the query args. I did argue here https://github.com/vegaprotocol/vega/pull/8114#discussion_r1169122618 that these are mostly not raw strings so SQL injection should not be possible, but it's a bit of a flimsy argument to be fair, we don't know what parameters we might add in the future...

So, although in my previous solution I only replaced the args (`$1`, `$2`...) with actual values and called it a day, I guess we can never be too safe.